### PR TITLE
V1.x ltr release

### PR DESCRIPTION
**Pull request by ychunxue**
*Issue #, if available:*

*Description of changes:*

Merging LTR changes to public KCL 1.x branch for LTR KCL release

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -6,7 +6,7 @@
   <artifactId>amazon-kinesis-client</artifactId>
   <packaging>jar</packaging>
   <name>Amazon Kinesis Client Library for Java</name>
-  <version>1.13.4-SNAPSHOT</version>
+  <version>1.13.4</version>
```
Comment (joshua-kim) [inline pom.xml:9]: Shouldn't this be 1.14.0?

File: pom.xml [hunk 2]
```diff
@@ -55,7 +55,7 @@
     <dependency>
       <groupId>com.google.protobuf</groupId>
       <artifactId>protobuf-java</artifactId>
-      <version>2.6.1</version>
+      <version>3.11.4</version>
```
Comment (ashwing) [inline pom.xml:58]: Why the protobuf change is not in master version?

**Comment by joshua-kim** [comment]
Need to change KinesisClientLibConfiguration L150 version

File: src/main/java/com/amazonaws/services/kinesis/clientlibrary/lib/worker/KinesisClientLibConfiguration.java [hunk 3]
```diff
@@ -89,6 +90,23 @@
      */
     public static final boolean DEFAULT_CLEANUP_LEASES_UPON_SHARDS_COMPLETION = true;
 
+    /**
```
Comment (ashwing) [inline src/main/java/com/amazonaws/services/kinesis/clientlibrary/lib/worker/KinesisClientLibConfiguration.java:93]: We need to update the user agent in this class

File: pom.xml (hunk 2 above)
Comment (joshua-kim) [reply pom.xml:58]: Weird, we do have it in v1.x master: https://github.com/awslabs/amazon-kinesis-client/commit/dd72cf96e5436b0002a5ebf363027d86e46f11b3

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,52 @@
...
+	| ----------------------------------- | ---------- | --------------------------------------------------------------------------------------------------------- |
+	| leaseCleanupIntervalMillis          | 1 minute   | Interval at which to run lease cleanup thread.                                                            |
+	| completedLeaseCleanupIntervalMillis | 5 minutes  | Interval at which to check if a lease is completed or not.                                                |
+	| garbageLeaseCleanupIntervalMillis   | 30 minutes | Interval at which to check if a lease is garbage (i.e trimmed past the stream's retention period) or not. |
+
+* Including an optimization to `KinesisShardSyncer` to only create leases for one layer of shards.
+* Changing default shard prioritization strategy to be `NoOpShardPrioritization` to allow prioritization of completed shards. Customers who are upgrading to this version and are reading from `TRIM_HORIZON` should continue using `ParentFirstPrioritization` while upgr    ading.
+* Upgrading version of AWS SDK to 2.13.X.
```
Comment (ashwing) [inline CHANGELOG.md:45]: This should align with 1.X

Comment (joshua-kim) [reply CHANGELOG.md:45]: Yeah, I left this as a TODO. This should say 1.XX (Also change the README).

**Comment by joshua-kim** [comment]
LGTM

File: pom.xml (hunk 2 above)
Comment (ychunxue) [reply pom.xml:58]: Not sure why it contains this diff. But double checked that we are merging to the KCL 1.x public branch. And this change does exist in v1.x master already

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,5 +1,52 @@
...
+	* Currently, the following shard filters are supported:
+		* `AT_TRIM_HORIZON` - the response includes all the shards that were open at `TRIM_HORIZON`.
+		* `AT_LATEST` - the response includes only the currently open shards of the data stream.
+	    * `AT_TIMESTAMP` - the response includes all shards whose start timestamp is less than or equal to the given timestamp and end timestamp is greater than or equal to the given timestamp or still open.
+	* `ShardFilter` is used when creating leases for an empty lease table to initialize leases for a snapshot of shards specified at `KinesisClientLibConfiguration#initialPositionInStreamExtended`.
+	* For more information about ShardFilter, see the [official AWS documentation on ShardFilter](https://docs.aws.amazon.com/kinesis/latest/APIReference/API_ShardFilter.html).
+
+* Introducing support for the `ChildShards` response of the `GetRecords` and the `SubscribeToShard` APIs to perform lease/shard synchronization that happens at `SHARD_END` for closed shards, allowing a KCL worker to only create leases for the child shards of the shard it finished processing.
```
Comment (ashwing) [inline CHANGELOG.md:16]: Remove all SubscribeToShard references

**Comment by ychunxue** [comment]
Merging LTR changes to public KCL 1.x branch. Preparing for KCL 1.14.0 release

**Pull request merged**